### PR TITLE
Better 517, and fix floating windows being behind

### PR DESCRIPTION
**Pull request by AethanFoot**
@VuiMuich I am sneaking another one in :wink:. This further improves #517. Also fixes the problem you mentioned in discord.

### CODE REVIEW

File: leftwm-core/src/handlers/window_move_handler.rs [hunk 1]
```diff
@@ -53,34 +55,34 @@ fn should_snap(window: &mut Window, workspace: &Workspace) -> bool {
...
+    if vec![
+        win_top - ws_top,
+        win_bottom - ws_bottom,
+        win_left - ws_left,
+        win_right - ws_right,
+    ]
+    .iter()
+    .map(|x| x.abs())
+    .any(|x| x < dist)
+    {
```
Comment (VuiMuich) [inline leftwm-core/src/handlers/window_move_handler.rs:85]: Also a very nice cleanup!

**Review by VuiMuich — APPROVED** [review]
Very nice, very sneaky 😃

File: leftwm-core/src/handlers/window_move_handler.rs [hunk 2]
```diff
@@ -53,34 +55,34 @@ fn should_snap(window: &mut Window, workspace: &Workspace) -> bool {
...
-    //get window sides
+    // Get window sides.
     let win_left = loc.x();
     let win_right = win_left + window.width();
     let win_top = loc.y();
     let win_bottom = win_top + window.height();
-    //check for conatins
+    // Check for conatins.
     let center_x = loc.x() + (window.width() / 2);
     let center_y = loc.y() + (window.height() / 2);
```
Comment (hertg) [inline leftwm-core/src/handlers/window_move_handler.rs:65]:
```suggestion
    let (center_x, center_y) = loc.center();
```

**Review by hertg — APPROVED** [review]

**Comment by VuiMuich** [comment]
closes #599

**Pull request merged**